subsys/bluetooth/audio/csis_client.c

```diff
@@ -140,6 +143,90 @@ static struct bt_csis *lookup_instance_by_set_info(const struct bt_csis_client_s
 	return NULL;
 }
 
+static struct bt_csis *get_next_active_instance(void)
+{
+	struct bt_csis_client_set_member *member;
+	struct bt_csis *inst;
+
+	member = active.members[active.members_handled];
+
+	inst =  lookup_instance_by_set_info(member, active.info);
+	if (inst == NULL) {
+		BT_DBG("Failed to lookup instance by set_info %p", active.info);
+	}
+
+	return inst;
+}
+
+static int member_rank_compare_asc(const void *m1, const void *m2)
+{
+	const struct bt_csis_client_set_member *member_1 =
+		*(const struct bt_csis_client_set_member **)m1;
+	const struct bt_csis_client_set_member *member_2 =
+		*(const struct bt_csis_client_set_member **)m2;
+	struct bt_csis *inst_1;
+	struct bt_csis *inst_2;
+
+	inst_1 = lookup_instance_by_set_info(member_1, active.info);
+	inst_2 = lookup_instance_by_set_info(member_2, active.info);
+
+	if (inst_1 == NULL) {
+		BT_ERR("inst_1 was NULL for member %p", member_1);
+		return 0;
+	}
+
+	if (inst_2 == NULL) {
+		BT_ERR("inst_2 was NULL for member %p", member_2);
+		return 0;
+	}
+
+	return inst_1->cli.rank - inst_2->cli.rank;
+}
+
+static int member_rank_compare_desc(const void *m1, const void *m2)
+{
+	/* If we call the "compare ascinding" function with the members
+	 * reversed, it will work as a descending comparison
+	 */
+	return member_rank_compare_asc(m2, m1);
+}
+
+static void active_members_store_ordered(struct bt_csis_client_set_member *members[],
+					 size_t count,
+					 const struct bt_csis_client_set_info *info,
+					 bool ascending)
+{
+	(void)memcpy(active.members, members, count * sizeof(members[0]));
+	active.members_count = count;
+	active.info = info;
+
+	qsort(active.members, count, sizeof(members[0]),
+	      ascending ? member_rank_compare_asc : member_rank_compare_desc);
+
+	if (IS_ENABLED(CONFIG_ASSERT) &&
+	    IS_ENABLED(CONFIG_BT_DEBUG_CSIS_CLIENT)) {
```

Comment:
I guess `CONFIG_BT_DEBUG_CSIS_CLIENT` is undesirable here, as when `CONFIG_ASSERT` is enabled you might want to assert, no matter if CSIS debug is enabled.

Reply:
Good point. This was a leftover from when I just did a `BT_ERR` instead.